lib/elixir/src/elixir_clauses.erl

```diff
@@ -54,6 +54,13 @@ warn_zero_length_guard({{'.', _, [erlang, '==']}, Meta,
                           "Prefer to pattern match on an empty list or use "
                           "\"~ts == []\" as a guard", [ArgString, ArgString]),
   elixir_errors:warn(?line(Meta), ?key(E, file), Message);
+warn_zero_length_guard({{'.', _, [erlang, '>']}, Meta,
+                        [{{'.', _, [erlang, length]}, _, [Arg]}, 0]}, E) ->
+  ArgString = 'Elixir.Macro':to_string(Arg),
+  Message = io_lib:format("\"length(~ts) > 0\" is discouraged since it has to "
```

Comment:
I would change the meaning of the sentence in an affirmative way:
> Is discouraged since it has to transverse the whole list just to see if it's empty